service/src/main/java/com/appirio/service/billingaccount/resources/BillingAccountResource.java

```diff
@@ -304,6 +305,82 @@ public ApiResponse getBillingAccountFees(@Auth AuthUser user, @PathParam("billin
             return ErrorHandler.handle(e, logger);
         }
     }
+    //SAROJ
+    /**
+     * Check Balance Amount.
+     * Check BillingAccount has availability to spend a given "amount"
+     *
+     * @param user
+     *            the currently logged in user
+     * @param queryParameter
+     *            the query parameters
+     * @param amount
+     *            the amount requested for locking from available budget ( budgetAmount - consumedAmount )
+     * @return the api response
+     */
+    
+    @GET
+    @Path("billing-accounts/{billingAccountId}/check-balance")
+    public ApiResponse checkBalanceFromBillingAccount(@Auth AuthUser user,
+            @PathParam("billingAccountId") Long billingAccountId,
+            @QueryParam("amount") Float amount) {
+        try {
+            checkAdmin(user, new String[] { READ_BILLING_ACCOUNT_SCOPE });
+            return MetadataApiResponseFactory.createResponse(billingAccountManager.checkBalance(billingAccountId, amount));// amount < availableAmount );
+        } catch (Exception e) {
+            return ErrorHandler.handle(e, logger);
+        }
+    }
+
+    /**
+     * Lock amount from the total BudgetAmount.
+     *
+     * @param user
+     *            the currently logged in user
+     * @param queryParameter
+     *            the query parameters
+     * @param lockAmount
+     *            the amount requested for locking from available budget ( availableAmount ) : Update the lockAmount
+     * @return the api response
+     */
+    //billing-accounts/{billingAccountId}/lock-amount
+    @PATCH
+    @Path("billing-accounts/{billingAccountId}/lock-amount")
+    public ApiResponse lockAmountFromBillingAccount(@Auth AuthUser user, @PathParam("billingAccountId") Long billingAccountId,
+            @Valid PostPutRequest<Float> lockAmount) {
+        try {
+            checkAdmin(user, new String[] { WRITE_BILLING_ACCOUNT_SCOPE });
+            return MetadataApiResponseFactory.createResponse(billingAccountManager.lockAmount(billingAccountId, lockAmount.getParam()));
+        } catch (Exception e) {
+            return ErrorHandler.handle(e, logger);
+        }
+    }
+
+    /**
+     * Cosume amount from the availableAmount
+     *
+     * @param user
+     *            the currently logged in user
+     * @param queryParameter
+     *            the query parameters
+     * @param consumedAmount
+     *            the amount requested for being consumed from available budget: Update the availableAmount
+     * @return the api response
+     */
+    
+    @PATCH
+    @Path("billing-accounts/{billingAccountId}/consume-amount")
+    public ApiResponse consumeAmountFromBillingAccount(@Auth AuthUser user, @PathParam("billingAccountId") Long billingAccountId,
+            @Valid PostPutRequest<ConsumedAmountDTO> consumedAmountDTO) {
```

Comment:
Let's add `challengeId: string` and `markup: float` to `ConsumedAmountDTO` - ideally these should be stored in DB for auditing, but we'll skip saving them to database for now and these values will only be used for publishing for now.